includes/api/class-wc-rest-reports-products-controller.php

```diff
@@ -31,4 +31,199 @@ class WC_REST_Reports_Products_Controller extends WC_REST_Reports_Controller {
 	 * @var string
 	 */
 	protected $rest_base = 'reports/products';
+
+	/**
+	 * Get items.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function get_items( $request ) {
+		$args = array();
+		$registered = array_keys( $this->get_collection_params() );
+		foreach ( $registered as $param_name ) {
+			if ( isset( $request[ $param_name ] ) ) {
+				$args[ $param_name ] = $request[ $param_name ];
+			}
+		}
+
+		$reports       = new WC_Reports_Products_Query( $args );
+		$products_data = $reports->get_data();
+
+		$data = array();
+
+		foreach ( $products_data as $product_data ) {
+			$item   = $this->prepare_item_for_response( (object) $product_data, $request );
+			$data[] = $this->prepare_response_for_collection( $item );
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare a report object for serialization.
+	 *
+	 * @param stdClass        $report  Report data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response
+	 */
+	public function prepare_item_for_response( $report, $request ) {
+		$data = get_object_vars( $report );
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+		$response->add_links( $this->prepare_links( $report, $request ) );
+
+		/**
+		 * Filter a report returned from the API.
+		 *
+		 * Allows modification of the report data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object           $report   The original report object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_report_products', $response, $report, $request );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param WC_Data         $object  Object data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return array                   Links for the given post.
+	 */
+	protected function prepare_links( $object, $request ) {
```

Comment:
`$request` is not used here, we include in some endpoints when required to fetch IDs or something else, but here we can remove it.

Reply:
Thanks, fixed.